### PR TITLE
apps/budgeting/proposal-list-item: adding features as moderator feedback e…

### PR DESCRIPTION
**Pull request by khamui**
…tc. and structure into smaller components

### CODE REVIEW

File: meinberlin/apps/budgeting/assets/ListItemBadges.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      {props.budget > 0 &&
+        <span className="label label--big">
+          {intComma(props.budget)}€
+        </span>}
+      {props.moderatorFeedback &&
+        <span
+          className={
+          'label' +
+          ' label--big' +
+          ' label--' + props.moderatorFeedback.toLowerCase() +
```
Comment (phillimorland) [inline meinberlin/apps/budgeting/assets/ListItemBadges.jsx:31]: nice! :) also just relaised we need to update this on the detail page oupps

File: meinberlin/apps/budgeting/assets/helpers.js [hunk 2]
```diff
@@ -14,7 +14,12 @@ const months = [
...
   django.gettext('December')
 ]
+
 export const toDate = (isodate) => {
   const d = new Date(isodate)
   return `${d.getDate()}. ${months[d.getMonth()]} ${d.getFullYear()}`
 }
+
+export const intComma = (number) => {
+  return Number(number).toLocaleString()
```
Comment (phillimorland) [inline meinberlin/apps/budgeting/assets/helpers.js:24]: is this adding a comma for cents? as we can only have whole numbers do we need it?

**Review by phillimorland — CHANGES_REQUESTED** [review]
Super cool! only thing is, when checking locally the margins between badges is off when more then 1 also the spacing between icons and numbers, once thats sorted it's good to go!

File: meinberlin/apps/budgeting/assets/helpers.js (hunk 2 above)
Comment (phillimorland) [reply meinberlin/apps/budgeting/assets/helpers.js:24]: maybe in js files we should consider having comments? as can't always tell, specially with no fuile name to help

Comment (khamui) [reply meinberlin/apps/budgeting/assets/helpers.js:24]: @phillimorland actually it is for the thousand separator (20.000 or 20,000). Works as the filter in django. I admit the name could be better but i named it how the template filter is named in django.

**Comment by khamui** [comment]
> Super cool! only thing is, when checking locally the margins between badges is off when more then 1 also the spacing between icons and numbers, once thats sorted it's good to go!

alright will check it!

**Comment by khamui** [comment]
@phillimorland i didn't know that span elements are that complicated. but it seems that if multiple spans are separated by line breaking in react, the whitespaces in between are getting removed. i found a solution which i implemented into a new span element because it really looks weird and screws the readability of our JSX. hit me up if you have a better idea.

File: meinberlin/apps/budgeting/assets/helpers.js (hunk 2 above)
Comment (phillimorland) [reply meinberlin/apps/budgeting/assets/helpers.js:31]: fab!

**Review by phillimorland — APPROVED** [review]
Really nice solution, I think we had this issue in comments too, will have to port the solution over at some point :)

**Pull request merged**